### ISSUE TITLE
Custom Loss vs Printed Loss

### ISSUE THREAD

**Issue by mshuaibii** [skorch-dev/skorch#535]
I'm not sure if there's someway to do this already - but I'm wanting to be able to train my model on a specified custom loss function but have skorch display a loss that's computed differently. For example: Model Loss = L2Norm + various regularization terms, etc. vs. Printed Loss = L2Norm. Is there an existing way to do this? I know I can manually do this by modifying the source code of `def train_step_single`  in `net.py` by adding lines after loss.backward() to compute my desired printed loss to be returned in history. Thanks!

**Comment by githubnemo** [comment]
You can add your own scores by adding an `EpochScoring` callback with a scoring function that computes the L2 norm of the model output (or whatever you like). Off the top of my head:

```python
from skorch.callbacks import EpochScoring

def l2score(net, X, y):
    y_pred = net.predict_proba(X)
    return np.norm(y_pred, p=2)

net = NeuralNet(
    my_module,
    callbacks=[EpochScoring(l2score)],
)
```

Does this help you?

**Comment by mshuaibii** [comment]
> You can add your own scores by adding an `EpochScoring` callback with a scoring function that computes the L2 norm of the model output (or whatever you like). Off the top of my head:
> 
> ```python
> from skorch.callbacks import EpochScoring
> 
> def l2score(net, X, y):
>     y_pred = net.predict_proba(X)
>     return np.norm(y_pred, p=2)
> 
> net = NeuralNet(
>     my_module,
>     callbacks=[EpochScoring(l2score)],
> )
> ```
> 
> Does this help you?

Ahh this seems like the solution - I'll give it a shot. Thanks!

**Comment by mshuaibii** [comment]
> You can add your own scores by adding an `EpochScoring` callback with a scoring function that computes the L2 norm of the model output (or whatever you like). Off the top of my head:
> 
> ```python
> from skorch.callbacks import EpochScoring
> 
> def l2score(net, X, y):
>     y_pred = net.predict_proba(X)
>     return np.norm(y_pred, p=2)
> 
> net = NeuralNet(
>     my_module,
>     callbacks=[EpochScoring(l2score)],
> )
> ```
> 
> Does this help you?

I tried copying this exact example but nothing seems to change?

**Comment by BenjaminBossan** [comment]
It looks like it should work. Can you provide your code to reproduce this?

**Comment by mshuaibii** [comment]
> It looks like it should work. Can you provide your code to reproduce this?

It seems that my issue comes from the `y_test = [self.target_extractor(y) for y in self.y_trues_]` my target is a list of 2 tensors. I'll try playing around with the structure and going from there.

**Comment by BenjaminBossan** [comment]
Sounds like something along the lines of `target_extractor=lambda y: to_numpy(y[0])` should work.

**Comment by BenjaminBossan** [comment]
@mshuaibii If your problem was solved, feel free to close the issue.
